model-optimizer/extensions/back/MarkNodesWithShapeValues.py

```diff
@@ -52,27 +53,48 @@ def get_operations_with_shape_inputs():
             'OneHot': [1],  # depth input
         }
 
-    def find_and_replace_pattern(self, graph: Graph):
-        shape_input_ops_map = self.get_operations_with_shape_inputs()
-
-        nodes_with_shape_inputs = []
+    @staticmethod
+    def get_nodes_with_shape_inputs(graph: Graph) -> List[Node]:
+        shape_accepting_ops = MarkNodesWithShapeValues.get_shape_accepting_ops()
+        shape_accepting_nodes = []
         for node in graph.get_op_nodes():
-            if node.soft_get('type') in shape_input_ops_map:
-                nodes_with_shape_inputs.append(node)
+            if node.soft_get('type') in shape_accepting_ops:
+                shape_accepting_nodes.append(node)
+        return shape_accepting_nodes
 
-        start_nodes = []
+    @staticmethod
+    def get_sources_for_nodes(nodes_with_shape_inputs: List[Node], filter_condition: Callable) -> List[Node]:
+        shape_accepting_ops = MarkNodesWithShapeValues.get_shape_accepting_ops()
+        sources = []
         for node in nodes_with_shape_inputs:
-            start_nodes.extend(
-                [node.in_port(port_idx).get_source().node for port_idx in shape_input_ops_map[node.type] if
-                 node.is_in_port_connected(port_idx)])
+            for port_idx in shape_accepting_ops[node.soft_get('type')]:
```

Comment:
add an assert here that `node.soft_get('type')` is in `shape_accepting_ops`. Also, rename the function to state that it works only for these nodes. Something like: `get_sources_for_shape_accepting_nodes`

Reply:
added assert.

Renamed function `get_shape_returning_start_nodes` preferred it over `get_sources_for_shape_accepting_nodes` since it returns not all sources but only node that will be used to start BFS (nodes on data path at port :0 and ShapeOf nodes will not be included)